File: Arrays/AnagramIndex.java
import java.util.* ;
import java.io.*; 
public class Solution {
	public static int[] anagramMapping(int n, int[] a, int[] b) {
		// Brute-force
		// int []storeIndex=new int[n];
		// for(int i=0;i<n;i++){
		// 	for(int j=0;j<n;j++){
		// 		if(a[i]==b[j]){
		// 			storeIndex[i]=j;
		// 		}
		// 	}
		// }
		// return storeIndex;

    //optimized approach

		 // Create a HashMap to store the value-to-index mapping of elements in array b
        HashMap<Integer, Integer> indexMap = new HashMap<>();
        for (int i = 0; i < n; i++) {
            indexMap.put(b[i], i);
        }
		// {
  		// 46: 0,
 	    // 12: 1,
 	    // 32: 2,
 	    // 28: 3
		// }

        // Create an array to store the result indices
        int[] result = new int[n];
        for (int i = 0; i < n; i++) {
            result[i] = indexMap.get(a[i]);
        }
      // a[0] = 12 -> indexMap.get(12) = 1 -> result[0] = 1
      // a[1] = 28 -> indexMap.get(28) = 3 -> result[1] = 3
      // a[2] = 46 -> indexMap.get(46) = 0 -> result[2] = 0
      // a[3] = 32 -> indexMap.get(32) = 2 -> result[3] = 2

        return result;
	}
}

File: Arrays/AnagramIndex.js
/**
 * Function to find anagram mappings
 * @param {number[]} a - First array
 * @param {number[]} b - Second array
 * @returns {number[]} - Mapped indices
 */
function anagramMapping(a, b) {
    const indexMap = new Map();

    // Populate the indexMap with elements from b
    b.forEach((value, index) => {
        indexMap.set(value, index);
    });

    // Create the result array by mapping elements from a to their indices in b
    return a.map(value => indexMap.get(value));
}

// Example usage
const a = [12, 28, 46, 32];
const b = [46, 12, 32, 28];

const result = anagramMapping(a, b);
console.log(result); // Output: [1, 3, 0, 2]
